src/ziggurat/kafka_consumer/consumer_driver.clj

```diff
@@ -1,22 +1,21 @@
 (ns ziggurat.kafka-consumer.consumer-driver
-  (:require [mount.core :refer [defstate]]
+  (:require [cambium.core :as clog]
+            [clojure.tools.logging :as log]
+            [mount.core :as mount :refer [defstate]]
+            [ziggurat.config :refer [ziggurat-config]]
             [ziggurat.kafka-consumer.consumer :as ct]
             [ziggurat.kafka-consumer.consumer-handler :as ch]
-            [ziggurat.config :refer [ziggurat-config]]
-            [clojure.tools.logging :as log]
-            [ziggurat.kafka-consumer.executor-service :refer :all]
-            [mount.core :as mount]
-            [ziggurat.metrics :as metrics]
-            [cambium.core :as clog])
+            [ziggurat.kafka-consumer.executor-service :refer [thread-pool DEFAULT_THREAD_COUNT]]
+            [ziggurat.metrics :as metrics])
   (:import (java.util.concurrent ExecutorService RejectedExecutionException)
            (org.apache.kafka.clients.consumer Consumer)))
 
 (defn- start-polling-with-consumer
   [consumer init-arg topic-entity consumer-config]
-  (let [message-poller (cast Runnable #(ch/poll-for-messages consumer (:handler-fn init-arg) topic-entity consumer-config))]
+  (let [message-poller (reify Runnable (run [_] (ch/poll-for-messages consumer (:handler-fn init-arg) topic-entity consumer-config)))]
```

Comment:
I'm getting ClassCast exceptions here. So instead of `cast`, used `reify` to create a Runnable object